### ISSUE TITLE
IDE0004 false positive for value tuple with nullable reference type

### ISSUE THREAD

**Issue by MaceWindu** [dotnet/roslyn#68881]
**Version Used**: 4.6.0-3.23259.8+c3cc1d0ceeab1a65da0217e403851a1e8a30086a

**Steps to Reproduce**:

```cs
public class Class1
{
	public void Test()
	{
		var _ = Execute(() =>
		{
			if (Random.Shared.Next(0, 2) == 1)
			{
				// IDE0004: cast is redundant
				return ((object?)null, 1);
				// after fix:
				// CS8619: Nullability of reference types in value of type '(object?, int)' doesn't match target type '(object, int)'
				//return (null, 1);
			}

			return (new object(), 2);
                        // with cast placed here alternatively, IDE0004 also triggered
			// return ((object?)new object(), 2);
		});
	}

	public T Execute<T>(Func<T> func) => func();
}

```

**Diagnostic Id**: IDE0004

**Expected Behavior**: no diagnostics

**Actual Behavior**: IDE0004 triggered with fix introducing CS8619

**Comment by sliekens** [comment]
I got very similar issues with nullable components of tuples in LINQ expressions. For example, this code generates a sequence of `(int, string?)`  and then filters the null values, so the result is a `(int, string)`, but the type system keeps fighting back.


``` csharp
public class Class1
{
    public void Linq()
    {
        (int num, string effect) data = ((int, string))Enumerable.Range(0, 100)
            .Select(num => (num, num % 2 == 0 ? "signal" : null))
            .First(data => data.Item2 is not null);
    }
}
```

I get TWO problems in my editor:

- Cast is redundant (IDE0004)
- Nullability of reference types in value of type '(int num, string?)' doesn't match target type '(int, string)' (CS8619)
